### DISCUSSION TITLE
VTX admin channel/band/send on switches

### DISCUSSION THREAD

**Discussion by genapauk** [ExpressLRS/ExpressLRS#2712] [Ideas]
Is there any possibility to bind VTX admin features (channel/band change and send new values to vrx and vtx) to my radiomaster switches, just like HT enable and HT start channel?

For example it's too long for me to push "sys" button, then find TBS agent lite or ExpressLRS lua script, then wait for options to render. It would be great if i could just bind these features to my switches, so i can switch between channels and bands with 1 and 2 button and send these values with button 3 (screenshots below).

It's not a feature request or idea future versions (but it would be great!!), i just wondering whether it's possible to do directly in code, because for now i have no idea where to start, would appreciate to get some clues or hints how and where i can do that.
I've already made some sort of my "custom" firmware versions of elrs with my own frequency domains in FHSS.cpp and unlocked sending vtx/vrx values with backpack even when drone is armed, so this feature would be a great addition for my work, thanks
![1](https://github.com/ExpressLRS/ExpressLRS/assets/30335671/1d2e3b09-8b32-4dbb-bd0d-da1b8833d317)
Note! Its has to be exactly vtx admin features, i know switching channels can be done with switches even using betaflight, but i need to do it using my backpack to switch both vrx and vtx simultaneously!

**Comment by alexkarlov** [comment]
I use Skyzone Steadyview X and have the same question)

**Comment by Z-Vladus** [comment]
+++
I agree, that will be awesome feature.

**Comment by samwidarr** [comment]
Hi, did you find a solution? I’m also very interested in the switches, if you find it, please share it, maybe with a link, thanks in advance.

**Comment by sebas1984x** [comment]
This would be really nice to have, switching channels in Seconds!

**Reply by Akceptor** [reply]
> This would be really nice to have, switching channels in Seconds!

Not really production ready but you can do it yourself https://github.com/Akceptor/VTx-Admin-on-Switch/tree/main

**Comment by Akceptor** [comment]
Definitely possible :)
https://www.youtube.com/watch?v=KGMa7ClkdXw
